### PR TITLE
Bilal/validate all links for EU, ROW and DIEL

### PR DESCRIPTION
**Pull request by bilalmajid-deriv**
Refactored logic for checkAllLinks.cy.js to create test for validateAllLinks.cy.js and complete the test. added recursive function for checkLink to traverse through all the links available on homepage as well as all the child pages which fall in the scope of Deriv.com. Adding Checking function to check for certain conditions as well before validating a link and to make sure the recursive function is not stuck in a loop.

The test takes a total time of approximately 7-10 minutes and validates all possible links and child links 
Added comments for each Function created 
two urls which are added in excluded visit and check links are to be removed after fix is provided for them.
Will remove the previous test file wip/checkAllLinks.cy.js once this is approved/merged.

### CODE REVIEW

**Review by markodonnell-deriv — CHANGES_REQUESTED** [review]
Looks good. Could you put the Clickup Test Case code in the describe?

**Comment by bilalmajid-deriv** [comment]
The Clickup Test ID has been added to describe. Thanks

**Comment by bilalmajid-deriv** [comment]
@markodonnell-deriv I have also added a check to verify if the visited page is not showing 404 page not found.

File: cypress/e2e/full/validateAllLinks.cy.js [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+                const currentLink = availableLink.prop('href')
+                checkLinks(currentLink)
+            })
+        }
+    }
+}
+
+
+
+describe('QATEST-96657 - Check URL in deriv.com', () => {
```
Comment (markodonnell-deriv) [inline cypress/e2e/full/validateAllLinks.cy.js:186]: I'm not seeing 96657 in the test case bank. Is this the squad task id?

**Review by markodonnell-deriv — CHANGES_REQUESTED** [review]

**Review by markodonnell-deriv — APPROVED** [review]

**Pull request merged**